src/main/java/de/tum/cit/aet/artemis/lecture/repository/SlideRepository.java

```diff
@@ -64,11 +67,20 @@ public interface SlideRepository extends ArtemisJpaRepository<Slide, Long> {
      *
      * @param slideId The ID of the slide to unhide
      */
-    @Transactional
-    default void unhideSlide(Long slideId) {
-        findById(slideId).ifPresent(slide -> {
-            slide.setHidden(null);
-            save(slide);
-        });
-    }
+    @Transactional // ok because of delete
+    @Modifying
+    @Query("""
+            UPDATE Slide s
+            SET s.hidden = NULL
+            WHERE s.id = :slideId
+            """)
+    void unhideSlide(@Param("slideId") Long slideId);
+
+    @Query("""
+            SELECT new de.tum.cit.aet.artemis.lecture.dto.SlideDTO(s.id, s.slideNumber, s.hidden, s.attachmentUnit.id)
+            FROM Slide s
+            WHERE s.attachmentUnit.id IN :attachmentUnitIds
+                AND s.hidden IS NULL OR s.hidden > CURRENT_TIMESTAMP()
```

Comment:
AND has a higher precedence than OR
```suggestion
                AND (s.hidden IS NULL OR s.hidden > CURRENT_TIMESTAMP())
```